applications/filestore-backup/templates/cronjobs.yaml

```diff
@@ -0,0 +1,97 @@
+kind: CronJob
+metadata:
+  name: create-backup
+spec:
+  schedule: {{ .Values.tool.backup.schedule | quote }}
+  successfulJobsHistoryLimit: 1
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          restartPolicy: Never
+          automountServiceAccountToken: true
+          {{- with .Values.tolerations }}
+          tolerations:
+{{ toYaml . | indent 12 }}
+          {{- end }}
+          {{- with .Values.tolerations }}
+          affinity:
+{{ toYaml . | indent 12 }}
+          {{- end }}
+          containers:
+          - name: create-backup
+            image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+            securityContext:
+              allowPrivilegeEscalation: false
+              runAsNonRoot: true
+              runAsUser: 1000
+              runAsGroup: 1000
+              capabilities:
+                drop:
+                  - all
+              readOnlyRootFilesystem: true
+            env:
+              - name: "RUBIN_GOOGLE_FILESTORE_TOOLS_PROJECT"
+                value: {{ .Values.global.gcpProjectId | quote }}
+              - name: "RUBIN_GOOGLE_FILESTORE_TOOLS_ZONE"
+                value: "{{ .Values.global.gcpRegion }}-{{ .Values.tool.zone }}"
+              - name: "RUBIN_GOOGLE_FILESTORE_TOOLS_INSTANCE"
+                value: {{ .Values.tool.instance | quote }}
+              - name: "RUBIN_GOOGLE_FILESTORE_TOOLS_SHARE_NAME"
+                value: {{ .Values.tool.fileShare | quote }}
+              {{- with .Values.tool.backup.debug }}
+              - name: "RUBIN_GOOGLE_FILESTORE_TOOLS_DEBUG"
+                value: "true"
+              {{- end }}
+            command: [ "create_backup" ]
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: purge-backup
+spec:
+  schedule: {{ .Values.tool.purge.schedule | quote }}
+  successfulJobsHistoryLimit: 1
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          restartPolicy: Never
+          automountServiceAccountToken: true
+          {{- with .Values.tolerations }}
+          tolerations:
+{{ toYaml . | indent 12 }}
+          {{- end }}
+          {{- with .Values.tolerations }}
+          affinity:
+{{ toYaml . | indent 12 }}
+          {{- end }}
+          containers:
+          - name: purge-backup
+            image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+            securityContext:
+              allowPrivilegeEscalation: false
+              runAsNonRoot: true
+              runAsUser: 1000
+              runAsGroup: 1000
+              capabilities:
+                drop:
+                  - all
+              readOnlyRootFilesystem: true
+            env:
+              - name: "RUBIN_GOOGLE_FILESTORE_TOOLS_PROJECT"
+                value: {{ .Values.global.gcpProjectId | quote }}
+              - name: "RUBIN_GOOGLE_FILESTORE_TOOLS_ZONE"
+                value: "{{ .Values.global.gcpRegion }}-{{ .Values.tool.zone }}"
```

Comment:
These values will only be set for GCP deployments which is going to produce syntactically correct but invalid configurations if someone mistakenly tries to deploy this on a non-GCP deployment. I think this should use the Helm `required` macro instead so that Helm throws an error if these values are not set.

Reply:
Done (for gcpProjectId), on the grounds that it's unlikely someone would specify one and not the other and the templating is uglier for a concatenated value.